frontend/src/components/Modal.tsx

```diff
@@ -0,0 +1,159 @@
+      onClose();
+    }
+  }
+
+  useEffect(() => {
+    // Focus the container div automatically to listen for "Escape" key press
+    if (drawerRef && isOpen) {
+      if (drawerRef?.current) drawerRef.current.focus();
+    }
+  }, [drawerRef, isOpen]);
+
+  function handleKeyPress(event: React.KeyboardEvent<HTMLDivElement>) {
+    // Close modal when user clicks "Escape" key
+    if (event.key === "Escape") {
+      onClose();
+    }
+  }
+
+  return (
+    <div
+      className={styles.modalDrawer}
+      ref={drawerRef}
+      onKeyDown={handleKeyPress}
+      onClick={handleMouseClick}
+      tabIndex={-1}
+      role="presentation"
+    >
+      {isOpen && (
+        <div className={styles.popupCanvas} ref={canvasRef}>
+          <div className={styles.popupWindow} ref={modalWindowRef}>
+            <button
+              className={styles.closeButtonContainer}
+              aria-label="Close Popup"
+              tabIndex={0}
+              type="button"
+            >
+              <img src={closeIcon} alt="Close Popup" ref={closeButtonRef} />
+            </button>
+            {children}
+          </div>
+        </div>
+      )}
+    </div>
+  );
+}
+
+export function CongratulationsModal(props: ModalProps) {
+  function returnToMainPage() {
+    window.location.href = "https://ccidc.org/";
+  }
+  return (
+    <Modal {...props}>
+      <div className={styles.modalChildren}>
+        <img className={styles.modalImage} src={checkMark} alt="Green Checkmark" />
+        <p className={styles.mainText}> Congratulations! </p>
+        <p className={styles.reviewText}>
+          Your application has been completed as is now under review with CCIDC.
+        </p>
+        <Button onClick={returnToMainPage}>Click here to return to Main Page</Button>
+      </div>
+    </Modal>
+  );
+}
+
+export function RequirementsNotMetModal(props: ModalProps) {
+  const navigate = useNavigate();
+  return (
+    <Modal {...props}>
+      <div className={styles.modalChildren}>
+        <img className={styles.modalImage} src={errorAlert} alt="Error Alert" />
+        <p className={styles.mainText}> Requirements Not Met </p>
+        <p className={styles.reviewText}>
+          Your application seems to be missing a necessary component. Please review the requirements
+          for each pathway{" "}
+          <button
+            onClick={() => {
+              navigate("/candidates");
+            }}
+            className={styles.modalLink + " " + styles.linkBlue}
+          >
+            here
+          </button>{" "}
+          and reapply when you have met the requirements for at least path one.
+        </p>
+        {/* TODO: route to correct page */}
+        <Button onClick={undefined}>Retake Prescreening Questions</Button>
+      </div>
+    </Modal>
+  );
+}
+
+type CompleteInOneSittingModalProps = {
+  path: 1 | 2 | 3 | 4;
+} & ModalProps;
+
+export function CompleteInOneSittingModal(props: CompleteInOneSittingModalProps) {
```

Comment:
Can we hook this up to the application page. Since application page is already in main and this popup isn't reliant on the forms